docs/tutorial/category_theory/category_theory_intro.md

```diff
@@ -0,0 +1,231 @@
+# An introduction to category theory in Lean
+
+This is an introduction to the basic usage of category theory (in the mathematical sense) in Lean.
+We cover how the basic theory of categories, functors and natural transformations is set up in Lean.
+Most of the below is not hard to read off from the files `category_theory/category.lean`,
+`category_theory/functor.lean` and `category_theory/natural_transformation.lean`, but it is a bit
+painful having to constantly have all three files open at once!
+
+
+First a word of warning. In `mathlib`, in the `/src` directory, there is a subdirectory called
+`category`. This is *not* where categories, in the sense of mathematics, are defined; it's for use
+by computer scientists. The directory we will be concerned with here is the `category_theory`
+subdirectory.
+
+The theory was largely developed by Scott Morrison, with the help of (who else to mention? **TODO**)
+
+## Overview
+
+A category is a collection of objects, and a collection of morphisms (also knows as arrows) between
+the objects. The objects and morphisms have some extra structure and satisfy some axioms -- see the
+[definition on Wikipedia](https://en.wikipedia.org/wiki/Category_(mathematics)#Definition) for
+details.
+
+One important thing to note is that a morphism in an abstract category may not be an actual map
+between two types. In particular, there is new notation `⟶` , typed as `\h` in VS Code, for a
+morphism, and in some fonts this arrow can be virtually indistinguishable from the standard function
+arrow `→` . (**TODO** -- perhaps say something about how to change font, or point to a resource? kmb
+would love to know this himself because he cannot tell the difference between the arrows on his
+laptop; one seems to be one pixel bigger than the other). In particular, usual functional
+programming techniques for function evaluations (e.g. writing `f x` if `x : X` and `f : X → Y`) are
+not in general possible when using morphisms in category theory (although sometimes this is possible
+(**TODO** is it?)).
+
+Another point of confusion can be universe issues. Following Lean's conventions for universe
+polymorphism, the objects of a category might live in one universe `u` and the morphisms in another
+universe `v`. Note that in many categories showing up in "set-theoretic mathematics", the morphisms
+between two objects often form a set, but the objects themselves may or may not form a set. In Lean
+this corresponds to the two possibilities `u=v` and `u=v+1`, known as `small_category` and
+`large_category` respectively. In the below we stick to the general polymorphic situation with `u`
+and `v` general universes.
```

Comment:
```suggestion
`large_category` respectively. 
In order to avoid proving the same statements for both small and large categories, we usually stick to the general polymorphic situation with `u`
and `v` independent universes, and we do this below.
```